enterprise/coderd/workspaceproxy.go

```diff
@@ -0,0 +1,155 @@
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/audit"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/httpapi"
+	"github.com/coder/coder/coderd/httpmw"
+	"github.com/coder/coder/codersdk"
+	"github.com/google/uuid"
+)
+
+// @Summary Create workspace proxy for organization
+// @ID create-workspace-proxy-for-organization
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Templates
+// @Param request body codersdk.CreateWorkspaceProxyRequest true "Create workspace proxy request"
+// @Param organization path string true "Organization ID"
+// @Success 201 {object} codersdk.WorkspaceProxy
+// @Router /organizations/{organization}/workspaceproxies [post]
+func (api *API) postWorkspaceProxyByOrganization(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx               = r.Context()
+		org               = httpmw.OrganizationParam(r)
+		auditor           = api.AGPL.Auditor.Load()
+		aReq, commitAudit = audit.InitRequest[database.WorkspaceProxy](rw, &audit.RequestParams{
+			Audit:   *auditor,
+			Log:     api.Logger,
+			Request: r,
+			Action:  database.AuditActionCreate,
+		})
+	)
+	defer commitAudit()
+
+	var req codersdk.CreateWorkspaceProxyRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	if err := validateProxyURL(req.URL, false); err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "URL is invalid.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	if err := validateProxyURL(req.WildcardURL, true); err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Wildcard URL is invalid.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	proxy, err := api.Database.InsertWorkspaceProxy(ctx, database.InsertWorkspaceProxyParams{
+		ID:             uuid.New(),
+		OrganizationID: org.ID,
+		Name:           req.Name,
+		DisplayName:    req.DisplayName,
+		Icon:           req.Icon,
+		// TODO: validate URLs
+		Url:         req.URL,
+		WildcardUrl: req.WildcardURL,
+		CreatedAt:   database.Now(),
+		UpdatedAt:   database.Now(),
+	})
+	if database.IsUniqueViolation(err) {
+		httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
+			Message: fmt.Sprintf("Workspace proxy with name %q already exists.", req.Name),
+		})
+		return
+	}
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
+
+	aReq.New = proxy
+	httpapi.Write(ctx, rw, http.StatusCreated, convertProxy(proxy))
+}
+
+// nolint:revive
+func validateProxyURL(u string, wildcard bool) error {
+	p, err := url.Parse(u)
+	if err != nil {
+		return err
+	}
+	if p.Scheme != "http" && p.Scheme != "https" {
+		return xerrors.New("scheme must be http or https")
+	}
+	if !(p.Path == "/" || p.Path == "") {
+		return xerrors.New("path must be empty or /")
+	}
+	if wildcard {
+		if !strings.HasPrefix(p.Host, "*.") {
+			return xerrors.Errorf("wildcard URL must have a wildcard subdomain (e.g. *.example.com)")
+		}
+	}
+	return nil
+}
```

Comment:
Use the same logic as we use in coderd for validating/parsing the subdomain wildcard. We have a function that checks it's good and creates a regex, if that passes then it should be good.